Exclude VCS directories by default

e.g. `.git`, `.hg`, `.svn` etc.

This panic occurs when scanning `.git` directories.
```
panic: decoding charset: IBM424_ltr: ianaindex: invalid encoding name

goroutine 1 [running]:
github.com/ianlewis/todos/internal/scanner.FromBytes({0xc012038d90, 0x68}, {0xc012030c00, 0xe6, 0x200})
        /usr/local/google/home/ianlewis/src/todos/internal/scanner/scanner.go:108 +0x733
github.com/ianlewis/todos/internal/scanner.FromFile(0xc000b048b0)
        /usr/local/google/home/ianlewis/src/todos/internal/scanner/scanner.go:81 +0xf3
github.com/ianlewis/todos/internal/walker.(*TODOWalker).scanFile(0xc00087db18, 0xc000b048b0)
        /usr/local/google/home/ianlewis/src/todos/internal/walker/walker.go:234 +0x2a
github.com/ianlewis/todos/internal/walker.(*TODOWalker).processFile(0xc00087db18, {0xc011c7b900, 0x36}, {0xc012038d90?, 0x4c8e2f?}, 0x9c4d40?)
        /usr/local/google/home/ianlewis/src/todos/internal/walker/walker.go:230 +0x74
github.com/ianlewis/todos/internal/walker.(*TODOWalker).walkFunc(0xc00087db18, {0xc011c7b900, 0x36}, {0x6c30b8, 0xc011c3a940}, {0x0?, 0x0?})
        /usr/local/google/home/ianlewis/src/todos/internal/walker/walker.go:183 +0x3d3
io/fs.walkDir({0x6c2100, 0xc0002719c0}, {0xc011c7b900, 0x36}, {0x6c30b8, 0xc011c3a940}, 0xc00087da28)
        /usr/local/google/home/ianlewis/opt/go/src/io/fs/walk.go:73 +0x78
io/fs.walkDir({0x6c2100, 0xc0002719c0}, {0xc011c36440, 0xf}, {0x6c30b8, 0xc000beaf80}, 0xc00087da28)
        /usr/local/google/home/ianlewis/opt/go/src/io/fs/walk.go:95 +0x2e6
io/fs.walkDir({0x6c2100, 0xc0002719c0}, {0xc001a212e0, 0xc}, {0x6c30b8, 0xc00020a3c0}, 0xc00087da28)
        /usr/local/google/home/ianlewis/opt/go/src/io/fs/walk.go:95 +0x2e6
io/fs.walkDir({0x6c2100, 0xc0002719c0}, {0xc00001aeb8, 0x4}, {0x6c30b8, 0xc000031f40}, 0xc00087da28)
        /usr/local/google/home/ianlewis/opt/go/src/io/fs/walk.go:95 +0x2e6
io/fs.walkDir({0x6c2100, 0xc0002719c0}, {0x6bf578, 0x1}, {0x6c3080, 0xc0002719d0}, 0xc00087da28)
        /usr/local/google/home/ianlewis/opt/go/src/io/fs/walk.go:95 +0x2e6
io/fs.WalkDir({0x6c2100, 0xc0002719c0}, {0x6bf578, 0x1}, 0xc0000a7a28)
        /usr/local/google/home/ianlewis/opt/go/src/io/fs/walk.go:122 +0xde
github.com/ianlewis/todos/internal/walker.(*TODOWalker).walkDir(...)
        /usr/local/google/home/ianlewis/src/todos/internal/walker/walker.go:134
github.com/ianlewis/todos/internal/walker.(*TODOWalker).Walk(0xc00087db18)
        /usr/local/google/home/ianlewis/src/todos/internal/walker/walker.go:117 +0x1a5
main.newTODOsApp.func1(0xc000031740)
        /usr/local/google/home/ianlewis/src/todos/internal/cmd/todos/app.go:107 +0x13a
github.com/urfave/cli/v2.(*Command).Run(0xc0002691e0, 0xc000031740, {0xc000016040, 0x2, 0x2})
        /usr/local/google/home/ianlewis/go/pkg/mod/github.com/urfave/cli/v2@v2.25.6/command.go:274 +0x9eb
github.com/urfave/cli/v2.(*App).RunContext(0xc0002fc000, {0x6c2e88?, 0xc00001a0c8}, {0xc000016040, 0x2, 0x2})
        /usr/local/google/home/ianlewis/go/pkg/mod/github.com/urfave/cli/v2@v2.25.6/app.go:332 +0x616
github.com/urfave/cli/v2.(*App).Run(...)
        /usr/local/google/home/ianlewis/go/pkg/mod/github.com/urfave/cli/v2@v2.25.6/app.go:309
main.main()
        /usr/local/google/home/ianlewis/src/todos/internal/cmd/todos/main.go:22 +0x45
exit status 2
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Leading whitespaces is now trimmed from TODOS in multi-line comments.
 - The `--include-hidden` option was replaced with the `--exclude-hidden`
   option and including hidden files was made the default.
+- An `--include-vcs` option was added and the VCS directories `.git`, `.hg`,
+  and `.svn` are skipped by default.
 
 ## Fixed in X.Y.Z
 

--- a/internal/cmd/todos/app.go
+++ b/internal/cmd/todos/app.go
@@ -64,6 +64,11 @@ func newTODOsApp() *cli.App {
 				DisableDefaultText: true,
 			},
 			&cli.BoolFlag{
+				Name:               "include-vcs",
+				Usage:              "Include version control directories (.git, .hg, .svn)",
+				DisableDefaultText: true,
+			},
+			&cli.BoolFlag{
 				Name:               "include-vendored",
 				Usage:              "Include vendored directories",
 				DisableDefaultText: true,
@@ -178,6 +183,8 @@ func walkerOptionsFromContext(c *cli.Context) (*walker.Options, error) {
 	o := walker.Options{}
 
 	o.IncludeHidden = !c.Bool("exclude-hidden")
+	o.IncludeVCS = c.Bool("include-vcs")
+	o.IncludeVendored = c.Bool("include-vendored")
 
 	outType := c.String("output")
 	outFunc, ok := outTypes[outType]

--- a/internal/cmd/todos/app_test.go
+++ b/internal/cmd/todos/app_test.go
@@ -311,6 +311,34 @@ func Test_walkerOptionsFromContext(t *testing.T) {
 				Paths:         []string{"."},
 			},
 		},
+		"include-vcs": {
+			args: []string{"--include-vcs"},
+			flags: []cli.Flag{
+				&cli.BoolFlag{
+					Name: "include-vcs",
+				},
+			},
+			expected: &walker.Options{
+				Config:        &todos.Config{},
+				IncludeHidden: true,
+				IncludeVCS:    true,
+				Paths:         []string{"."},
+			},
+		},
+		"include-vendored": {
+			args: []string{"--include-vendored"},
+			flags: []cli.Flag{
+				&cli.BoolFlag{
+					Name: "include-vendored",
+				},
+			},
+			expected: &walker.Options{
+				Config:          &todos.Config{},
+				IncludeHidden:   true,
+				IncludeVendored: true,
+				Paths:           []string{"."},
+			},
+		},
 		"paths": {
 			args: []string{"/path/to/code"},
 			expected: &walker.Options{

--- a/internal/walker/walker.go
+++ b/internal/walker/walker.go
@@ -59,6 +59,9 @@ type Options struct {
 	// paths are always processed if there are specified explicitly in `paths`.
 	IncludeVendored bool
 
+	// IncludeVCS indicates that VCS paths (.git, .hg, .svn, etc.) should be included.
+	IncludeVCS bool
+
 	// Paths are the paths to walk to look for TODOs.
 	Paths []string
 }
@@ -202,6 +205,10 @@ func (w *TODOWalker) processDir(path, fullPath string) error {
 		return fs.SkipDir
 	}
 
+	if !w.options.IncludeVCS && isVCS(fullPath) {
+		return fs.SkipDir
+	}
+
 	// NOTE: linguist only matches paths with a *nix path separators.
 	// TODO(github.com/ianlewis/linguist/issues/1): Update when linguist supports Windows.
 	fullPath = strings.ReplaceAll(fullPath, string(os.PathSeparator), "/")
@@ -279,4 +286,10 @@ func (w *TODOWalker) handleErr(prefix string, err error) error {
 		}
 	}
 	return nil
+}
+
+// isVCS returns whether the path is a vcs path. Should only be called on directories.
+func isVCS(path string) bool {
+	basePath := filepath.Base(path)
+	return basePath == ".git" || basePath == ".hg" || basePath == ".svn"
 }

